src/algebra/ring/basic.lean

```diff
@@ -116,6 +116,18 @@ class semiring (α : Type u) extends add_comm_monoid α, monoid_with_zero α, di
 section semiring
 variables [semiring α]
 
+@[priority 200] -- see Note [lower instance priority]
+instance semiring.to_non_unital_non_assoc_semiring : non_unital_non_assoc_semiring α :=
+{ ..‹semiring α› }
+
+@[priority 200] -- see Note [lower instance priority]
+instance semiring.non_unital_semiring : non_unital_semiring α :=
+{ ..‹semiring α› }
+
+@[priority 200] -- see Note [lower instance priority]
+instance semiring.non_assoc_semiring : non_assoc_semiring α :=
+{ ..‹semiring α› }
```

Comment:
Isn't the default priority for `extends` 100 (eg, see `semiring.to_add_comm_monoid`). If there's a reason for using a different 200 here, can you include it in a comment?

Reply:
Oh sorry I was being stupid. These priority settings are here only because I was mindlessly copying from the definition of `algebra.semimodule`. I've just dropped them. If work is quiet this morning I'll try and resolve the build failures too.

Reply:
To be clear; there _should_ be priority settings here, because you're trying to emulate `extends`, and `extends` puts priority 100 on conversion between the derived and base class. My comment was that choosing `200` was odd, not that the presence of `priority` at all was odd.

Reply:
I've literally just realised this :-)

Reply:
Doing this all too quickly. Fix incoming!